opentelemetry-tracing/opentelemetry-go/gorilla-mux/client/main.go

```diff
@@ -0,0 +1,77 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"time"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+
+	"github.com/signalfx/splunk-otel-go/distro"
+)
+
+func main() {
+	// handle CTRL+C gracefully
+	ctx, stop := signal.NotifyContext(context.Background(), os.Interrupt)
+	defer stop()
+
+	// initialize Splunk OTel distro
+	sdk, err := distro.Run()
+	if err != nil {
+		panic(err)
+	}
+	defer func() {
+		if err := sdk.Shutdown(context.Background()); err != nil {
+			panic(err)
+		}
+	}()
+
+	// instrument http.Client
+	client := &http.Client{Transport: otelhttp.NewTransport(http.DefaultTransport)}
+
+	for {
+		select {
+		case <-ctx.Done():
+			return
+		case <-time.After(time.Second):
+			call(ctx, client)
+		}
+	}
+}
+
+func call(ctx context.Context, client *http.Client) {
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:8080/hello", http.NoBody)
+	if err != nil {
+		panic(err)
```

Comment:
Returning these errors to the main `for` loop and printing them there would allow for retryable issues instead of having to restart the application.

```suggestion
func call(ctx context.Context, client *http.Client) error {
	req, err := http.NewRequestWithContext(ctx, http.MethodGet, "http://localhost:8080/hello", http.NoBody)
	if err != nil {
		return err
```

Reply:
`http.NewRequestWithContext` would return error only if the arguments are invalid
even if the URL would be configurable then the user would have to restart the application